scanpy/logging.py

```diff
@@ -159,28 +165,56 @@ def print_header(*, file=None):
     )
 
 
-def print_versions(*, file=None):
-    """Print print versions of imported packages"""
-    if file is None:  # Inform people about the behavior change
-        warning('If you miss a compact list, please try `print_header`!')
-    stdout = sys.stdout
-    try:
-        buf = sys.stdout = io.StringIO()
-        sinfo(
-            dependencies=True,
-            excludes=[
-                'builtins',
-                'stdlib_list',
-                'importlib_metadata',
-                # Special module present if test coverage being calculated
-                # https://gitlab.com/joelostblom/sinfo/-/issues/10
-                "$coverage",
-            ],
-        )
-    finally:
-        sys.stdout = stdout
-    output = buf.getvalue()
-    print(output, file=file)
+def _run_session_info(rich: bool) -> Optional[HTML]:
+    return session_info.show(
+        dependencies=True,
+        html=rich,
+        excludes=[
+            'builtins',
+            'stdlib_list',
+            'importlib_metadata',
+            # Special module present if test coverage being calculated
+            # https://gitlab.com/joelostblom/session_info/-/issues/10
+            "$coverage",
+        ],
+    )
+
+
+def print_versions(*, file: Optional[IO[str]] = None, rich: Optional[bool] = None):
+    """\
+    Print versions of imported packages, OS, and jupyter environment.
+
+    Parameters
+    ----------
+    file
+        File to write output to. The default `None` means writing to `stdout`
+    rich
+        Rich HTML output. Defaults to `True` only if `file` is `None`
+        and this is called from a jupyter notebook.
+    """
+    in_notebook = 'jupyter_core' in sys.modules
```

Comment:
Don't we already have a way of doing this? `ScanpyConfig._is_run_from_ipython()` I think?